packages/jupyter-ai/jupyter_ai/actors/autonotebook.py

```diff
@@ -0,0 +1,235 @@
+                "title",
+                "content",
+                "code_so_far"
+            ],
+        )
+        return cls(prompt=prompt, llm=llm, verbose=verbose)
+
+def generate_code(outline, llm=None, verbose=False):
+    if llm is None:
+        llm = ChatOpenAINewProvider(model_id='gpt-3.5-turbo')
+    chain = NotebookSectionCodeChain.from_llm(llm=llm, verbose=verbose)
+    code_so_far = []
+    for section in outline['sections']:
+        code = chain.predict(
+            description=outline['description'],
+            title=section['title'],
+            content=section['content'],
+            code_so_far='\n'.join(code_so_far)
+        )
+        section['code'] = improve_code(code, llm=llm, verbose=verbose)
+        code_so_far.append(code)
+    return outline
+
+class NotebookSummaryChain(LLMChain):
+    """Chain to generates tasks."""
+
+    @classmethod
+    def from_llm(cls, llm: BaseLLM, verbose: bool=False) -> LLMChain:
+        task_creation_template = (
+            "Create a markdown summary for a Jupyter notebook with the following content."
+            " The summary should consist of a single paragraph.\n"
+            "Content:\n{content}"
+        )
+        prompt = PromptTemplate(
+            template=task_creation_template,
+            input_variables=[
+                "content",
+            ],
+        )
+        return cls(prompt=prompt, llm=llm, verbose=verbose)
+
+class NotebookTitleChain(LLMChain):
+    """Chain to generates tasks."""
+
+    @classmethod
+    def from_llm(cls, llm: BaseLLM, verbose: bool=False) -> LLMChain:
+        task_creation_template = (
+            "Create a short, few word, descriptive title for a Jupyter notebook with the following content.\n"
+            "Content:\n{content}"
+        )
+        prompt = PromptTemplate(
+            template=task_creation_template,
+            input_variables=[
+                "content",
+            ],
+        )
+        return cls(prompt=prompt, llm=llm, verbose=verbose)
+
+def generate_title_and_summary(outline, llm=None, verbose=False):
+    if llm is None:
+        llm = ChatOpenAINewProvider(model_id='gpt-3.5-turbo')
+    summary_chain = NotebookSummaryChain.from_llm(llm=llm, verbose=verbose)
+    title_chain = NotebookTitleChain.from_llm(llm=llm, verbose=verbose)
+    summary = summary_chain.predict(content=outline)
+    title = title_chain.predict(content=outline)
+    outline['summary'] = summary
+    outline['title'] = title.strip('"')
+    return outline
+
+def create_notebook(outline):
+    nb = nbformat.v4.new_notebook()
+    markdown_cell = nbformat.v4.new_markdown_cell('# ' + outline['title'])
+    nb['cells'].append(markdown_cell)
+    markdown_cell = nbformat.v4.new_markdown_cell('## Introduction')
+    nb['cells'].append(markdown_cell)
+    disclaimer = f"This notebook was created by [Jupyter AI](https://github.com/jupyterlab/jupyter-ai) with the following prompt:\n\n> {outline['description']}"
```

Comment:
I noticed that when I run `/autonotebook` in Jupyter AI, the output notebook doesn't have the same description here as what I typed in.

For example, when I run:

```
/autonotebook Create a sample data set and use matplotlib to plot a bar chart using the data set
```

The resulting notebook says:
```
This notebook was created by [Jupyter AI](https://github.com/jupyterlab/jupyter-ai) with the following prompt:

> Creating a bar chart using sample data set with matplotlib
```

Reply:
Good, catch fixing this.